### PR TITLE
Fixed Bug

### LINKED ISSUE #40: IDOR
# Task
Hey contributors, the first **cybersec** issue is here.
The web application doesn't seem safe for **private** images. We need you to find the bug and patch it. 

# Steps
- Clone the repository after forking it
- Find the bug and patch it
- Create a PR to the **'main'** branch

# Rules
This is an **open to all** issue. So no need to **claim** it, just making a *proper PR* would be enough.

We follow **FCFS**, **F**irst **C**ome **F**irst **S**erve

We will give points to the contributor who makes **the first valid PR**
However, if we aren't satisfied with it we may reject it and consider other PRs.

*So keep your modifications clean and meaningful.*

Follow [README.md](https://github.com/opencodeiiita/PhotoStore/blob/main/README.md) to get started.

**NOTE:** this Issue is restricted to the **main** branch
You are requested to create a Pull request for that branch (main) only

### PR DESCRIPTION
**Pull request by criticic**
Issue: 40 

#### Short description of what this resolves:

Fixed Bug where unauthorized user could access any photo by predicting id

Fixes #40

### CODE REVIEW

**Comment by criticic** [comment]
@Srujana2505 Please take a look.

File: app.py [hunk 1]
```diff
@@ -236,6 +243,9 @@ def api_image_get(id):
     if not filename:
         return "", 404
 
+    if not owner == username or not public:
```
Comment (Mitul16) [inline app.py:246]: There is a problem here, how come the owner see their private images? :eyes:

Comment (criticic) [reply app.py:246]: For owner's private files:
Owner = yes &public = no
Wouldn't it work as expected? Or do you mean something else

Comment (criticic) [reply app.py]: @Mitul16 I think it should be resolved now.

**Comment by Mitul16** [comment]
@criticic Try to provide a POC as well, I am allowing this one as the Issue wasn't too big :+1:

**Comment by Mitul16** [comment]
> something like this?

Yes, that would work as well. *Simple enumeration!*
If you take a look at the **filename** when you use the **download** button
You will find `<USERNAME>-<HASH>.ext`, it can be used to build a profile of the acquired data :smile:

**Pull request merged**